Tribe Embeds Updates (#17)
moderntribe/tribe-embeds: readme.md

This repo is setup to use the [WP CLI dist-archive](https://developer.wordpress.org/cli/commands/dist-archive/) command.  To build the zip file for the make sure you have the dist-archive command package installed and run `wp dist-archive .` form the root folder. The zip file will be created one folder back form the root folder.

### Vimeo

In order to make Vimeo provider thumbs work correctly create an access token https://help.vimeo.com/hc/en-us/articles/12427789081745-How-to-generate-a-personal-access-token
Use Tribe Embeds Settings page to store token via DB or set `VIMEO_ACCESS_TOKEN` in you wp-config.php

## Hooks (filters & actions)

These are the **public** extension points intended for themes/plugins to customize behavior. Names and arguments are considered part of the API.

### Filters

> #### `tribe-embeds_video_provider`

Reviewer: I get the dash use in `tribe-embeds` but it feels a little uncomfortable also to mix dash and underscore.

Author: Replaced with dash version

    - `$block` — full Gutenberg block array (name, attributes, innerBlocks, etc.)
- **Return:** A `Provider` instance or `null` to skip.
- **Example:**
```php
add_filter( 'tribe-embeds_video_provider', function ( $provider, $video_url_data, $block ) {
  // Force our custom provider for a specific host or path
  if ( isset( $video_url_data['host'] ) && $video_url_data['host'] === 'videos.example.com' ) {
      return new \Tribe\Tribe_Embed\Providers\Example_Provider( $video_url_data );
  }
  return $provider;
}, 10, 3 );
```

#### `tribe-embeds_allowed_provider_hosts`

Expand or restrict the whitelist of hostnames that can be handled by built-in or custom providers.

- **Signature:** `apply_filters( 'tribe-embeds_allowed_provider_hosts', $allowed_hosts, $host )`
- **Args:**
  - `$allowed_hosts` — array of allowed host strings (e.g. ['youtube.com', 'youtu.be', 'vimeo.com', 'dailymotion.com'])
  - `$host` — the currently detected host
- **Return:** Modified array of allowed hosts.
- **Example:**
```php
add_filter( 'tribe-embeds_allowed_provider_hosts', function ( array $hosts ) {
    $hosts[] = 'videos.example.com';
    return $hosts;
}, 10 );
```

> #### `tribe_embeds_video_provider`

Reviewer: This and the filter below are duplicated from above but with the switch to all underscore. I think this readme just needs cleaned up a bit.

- **Args:**
    - `null|<Provider object>` — if object provided resolves immediately and return provided object 
    - `$video_url_data` — embed video ulr
    - `$block` — current embed block data
- **Return:** `null` or provided provider class
- **Example:**
```php
add_filter( 'tribe_embeds_video_provider', function ( $obj, $video_url_data, $block ) {
    // Note: $video_url_data has parsed video url. Provider accepts url string 
    $provider = new CustomProvider( $video_url );
    
    return $provider;
}, 10 );
```

#### `tribe_embeds_allowed_provider_hosts_<slug>`

Adjust allowed hosts for provider

- **Signature:** `apply_filters( 'tribe_embeds_allowed_provider_hosts_' . $slug, $base, $provider_class );`
- **Args:**
    - `$base` — list of allowed hosts
    - `$provider_class` — current provider class

#### `tribe_embeds_allowed_provider_hosts`

Adjust allowed hosts for provider

- **Signature:** `apply_filters( 'tribe_embeds_allowed_provider_hosts', $by_provider, $provider_class );`
- **Args:**
    - `$by_provider` — List of hosts returned from `tribe_embeds_allowed_provider_hosts_<slug>`
    - `$provider_class` — current provider class

> #### `tribe_embeds_image_sizes_<slug>`

Reviewer: I'm a little confused on the intent and difference in the filter here and below.

Author: This specific for provider. The below one is global version. In case you want catch all items one by one and not only the specific one

Get image sizes for a provider

- **Signature:** `apply_filters( 'tribe_embeds_image_sizes_' . $slug, $base, $provider_class );`
- **Args:**
    - `$base` — list of image sizes
    - `$provider_class` — current provider class

#### `tribe_embeds_image_sizes`

Get image sizes for a provider

- **Signature:** `apply_filters( 'tribe_embeds_image_sizes', $by_provider, $provider_class );`
- **Args:**
    - `$by_provider` — list of image sizes `tribe_embeds_image_sizes_<slug>`
    - `$provider_class` — current provider class

#### `tribe_embeds_provider_classes`

Allow external override of provider class list

- **Signature:** `apply_filters( 'tribe_embeds_provider_classes', $provider_classes ?: $defaults );`
- **Args:**
    - `$provider_classes` — list of existing providers classes

#### `tribe-embed_<video-provider>_video_thumbnail_url`

> Allows to adjust image data for each provider. Use slug instead of `<video-provider>` e.g `tribe-embed_wistia_video_thumbnail_url`

Reviewer: ```suggestion
Allows adjusting image data for each provider. Use slug instead of `<video-provider>` e.g `tribe-embed_wistia_video_thumbnail_url`
```

- **Signature:** `apply_filters( 'tribe-embed_wistia_video_thumbnail_url', $image_data, $video_id )`
- **Args:**
    - `$image_data` — Thumbnail image data
    - `$video_id` — current video id
- **Return:** Video thumbnail image data.

####  `tribe_embeds_facade_html`

> Fires and action to which the new block markup is added too.

Reviewer: ```suggestion
Fires an action with the new block markup attached.
```
